zingolib/src/wallet/tx_map.rs

```diff
@@ -1,25 +1,26 @@
 //! This mod should be called tx_map_and_maybe_trees.rs. it contains
-//! struct TxMapAndMaybeTrees
-//! implementations for TxMapAndMaybeTrees
-//! associated types for TxMapAndMaybeTrees that have no relevance elsewhere.
+//! struct TxMap
+//! implementations for TxMap
+//! associated types for TxMap that have no relevance elsewhere.
 
 use crate::{
     data::witness_trees::WitnessTrees,
     wallet::transaction_records_by_id::{
         trait_inputsource::InputSourceError, TransactionRecordsById,
     },
 };
+use spending_data::SpendingData;
 use std::{fmt::Debug, sync::Arc};
 use thiserror::Error;
 use zcash_primitives::legacy::TransparentAddress;
 
 /// HashMap of all transactions in a wallet, keyed by txid.
 /// Note that the parent is expected to hold a RwLock, so we will assume that all accesses to
 /// this struct are threadsafe/locked properly.
-pub struct TxMapAndMaybeTrees {
+pub struct TxMap {
```

Comment:
I think this name is less clear than TxMapAndMaybeTrees. What's the reasoning here?

Reply:
is Map the same as sById?

Reply:
I think so, yes

Reply:
okay, darn it, im sorry, i guess i picked the wrong name again. i regret it, darn oops. sorry. maybe we can revert it later